hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/store/ReadableStoreFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.store;
+
+import com.swirlds.state.HederaState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Factory for all readable stores. It creates new readable stores based on the {@link HederaState}.
```

Comment:
Interesting that implementations should create new readable stores based on `HederaState`, but this is the only mention of state in this file. Perhaps the second sentence belongs in the implementation class?